dbt/adapters/synapse/connections.py

```diff
@@ -0,0 +1,30 @@
+from contextlib import contextmanager
+
+import pyodbc
+import os
+import time
+import struct
+
+import dbt.exceptions
+import dbt.adapters.sqlserver.connections as sqlserverconn
+from dbt.adapters.sqlserver import SQLServerConnectionManager
+from dbt.adapters.sqlserver import SQLServerCredentials
+
+from dbt.logger import GLOBAL_LOGGER as logger
+
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class SynapseCredentials(SQLServerCredentials):
+    encrypt: Optional[bool] = True
+    trust_cert: Optional[bool] = True
+    
+    @property
+    def type(self):
+        return "synapse"
+
+
+class SynapseConnectionManager(SQLServerConnectionManager):
+    TYPE = "synapse"
```

Comment:
@jtcohen6 I'm guess I have to re-implement all of the methods again, even if they're copied and pasted directly?

Reply:
answering my own question. looks like yes -- I do.

Reply:
@swanderz Is that right? I was under the impression that classmethods from `SQLServerConnectionManager` are automatically inherited by `SynapseConnectionManager`, unless you need to reimplement them explicitly.

E.g. [`RedshiftConnectionManager`](https://github.com/fishtown-analytics/dbt/blob/10b33ccaf6a4db5fd89a80709579bbb46fe8de09/plugins/redshift/dbt/adapters/redshift/connections.py#L68) does not reimplement [`open`](https://github.com/fishtown-analytics/dbt/blob/10b33ccaf6a4db5fd89a80709579bbb46fe8de09/plugins/postgres/dbt/adapters/postgres/connections.py#L73), [`cancel`](https://github.com/fishtown-analytics/dbt/blob/10b33ccaf6a4db5fd89a80709579bbb46fe8de09/plugins/postgres/dbt/adapters/postgres/connections.py#L123), [`get_response`](https://github.com/fishtown-analytics/dbt/blob/10b33ccaf6a4db5fd89a80709579bbb46fe8de09/plugins/postgres/dbt/adapters/postgres/connections.py#L151) from `PostgresConnectionManager`. This is one of the biggest benefits of having `dbt-synapse` inherit from `dbt-sqlserver`